Analysis/src/EventInfo.cpp

```diff
@@ -62,48 +60,33 @@ const analysis::LepCandidate& EventInfoBase::GetSecondLeg()
 {
     Lock lock(*mutex);
     if(!leg2) {
-        tuple_leg2 = std::make_shared<ntuple::TupleLepton>(*event, GetLegIndex(2));
+        tuple_leg2 = std::make_shared<ntuple::TupleLepton>(event_candidate->GetEvent(), GetLegIndex(2));
```

Comment:
GetSecondLeg shouldn't create any new objects. Use the one created in event_candidate.